pkg/api/v1/router_instance_metadata.go

```diff
@@ -381,6 +383,39 @@ func handleDeleteRequest(c *gin.Context, r *Router, instanceID string, metadata
 		deleteInstanceIPs = (metadata == nil)
 	}
 
+	deleteSuccess := false
+	maxDeleteRetries := viper.GetInt("crdb.max_retries")
+	dbRetryInterval := viper.GetDuration("crdb.retry_interval")
+
+	for i := 0; i <= maxDeleteRetries; i++ {
+		err := performDeletions(c, r, instanceID, metadata, userdata, deleteMetadata, deleteUserdata, deleteInstanceIPs)
+		if err == nil {
+			if i > 0 {
+				r.Logger.Sugar().Info("DB delete transaction for instance ", instanceID, " successful on retry attempt #", i)
+			}
+
+			deleteSuccess = true
+
+			break
+		}
+
+		// Exponential backoff would be overkill here, but adding a bit of jitter
+		// to sleep a short time is reasonable
+		jitter := time.Duration(rand.Int63n(int64(dbRetryInterval)))
+		time.Sleep(jitter)
```

Comment:
similar comments here as the upserter, around maybe using an `else` block and changing the conditional in the loop a little

Reply:
Updated in both the delete retries as well as the upsert retries loops.